pkg/controllers/machine/lifecycle/registration.go

```diff
@@ -69,17 +69,27 @@ func (r *Registration) Reconcile(ctx context.Context, machine *v1alpha5.Machine)
 	metrics.MachinesRegisteredCounter.With(prometheus.Labels{
 		metrics.ProvisionerLabel: machine.Labels[v1alpha5.ProvisionerNameLabelKey],
 	}).Inc()
+	metrics.NodesCreatedCounter.With(prometheus.Labels{
+		metrics.ProvisionerLabel: machine.Labels[v1alpha5.ProvisionerNameLabelKey],
+	}).Inc()
 	return reconcile.Result{}, nil
 }
 
 func (r *Registration) syncNode(ctx context.Context, machine *v1alpha5.Machine, node *v1.Node) error {
 	stored := node.DeepCopy()
 	controllerutil.AddFinalizer(node, v1alpha5.TerminationFinalizer)
-	lo.Must0(controllerutil.SetOwnerReference(machine, node, scheme.Scheme))
+
 	// Remove any provisioner owner references since we own them
 	node.OwnerReferences = lo.Reject(node.OwnerReferences, func(o metav1.OwnerReference, _ int) bool {
 		return o.Kind == "Provisioner"
 	})
+	node.OwnerReferences = append(node.OwnerReferences, metav1.OwnerReference{
```

Comment:
Are we supposed to be adding in the machine.OwnerReferences here too or is this added on creation?